go/service/assets/api_op_ListAssets.go

```diff
@@ -15,11 +17,27 @@ type AssetOutput struct {
 	AssetProperties map[string]string `json:"properties"`
 }
 
-func (c *Client) ListAssets() ([]AssetOutput, error) {
-	endpoint := fmt.Sprintf("%s/assets", *c.Addresses.Management)
+func (c *Client) ListAssets(args ...apivalidator.QueryInput) ([]AssetOutput, error) {
+	var queryInput apivalidator.QueryInput
+
+	if len(args) > 0 {
+		queryInput = args[0]
```

Comment:
Just a question here: 
why do we take only one `queryInput`? If the connector API accepts only one query Input and not an array, I suggest updating the `ListAssets` parameter to a `*apivalidator.QueryInput` rather than an array of `apivalidator.QueryInput`. wdyt?

Reply:
That line of code checks if we have a `queryInput` in the first place because it's optional. As you mentioned, the connector API expects only one query max.
The only reason why I went with this approach is to be able to call `ListAssets()` this way rather than `ListAssets(nil)` in case we don't have a query. [You can see pictures above of how I coded it previously]
![343574294_583671690412259_5966455409479978209_n](https://github.com/Think-iT-Labs/edc-connector-client/assets/61057075/d7e4643b-4e55-4cc5-bf61-3c1c4c67a359)

![344762782_6066875723439747_4831448852812178306_n](https://github.com/Think-iT-Labs/edc-connector-client/assets/61057075/bb4e6312-350e-4b5c-8b6d-ab9379abe828)

The approach is very biased because it follows the way we do it in Javascript 😅 But I'm open to change it too.

Reply:
Yes makes more sense now, I see that your approach is better now. Thank you 🙏.